llvm/lib/Analysis/DependenceAnalysis.cpp

```diff
@@ -3991,6 +4003,23 @@ DependenceInfo::depends(Instruction *Src, Instruction *Dst,
     if (CompleteLoops[II])
       Result.DV[II - 1].Scalar = false;
 
+  // Set the distance to zero if the direction is EQ.
+  // TODO: Ideally, the distance should be set to 0 immediately simultaneously
+  // with the corresponding direction being set to EQ.
+  for (unsigned II = 1; II <= Result.getLevels(); ++II) {
+    if (Result.getDirection(II) == Dependence::DVEntry::EQ)
```

Comment:
Is it only that Distance is nullptr, or are there cases where it is non-zero but direction is still EQ?

In the former case, could you set the direction only when Distance is nullptr and assert if not `SCEV::isZero`?

In the latter case, the would be something quite wrong. Why prioritize the Direction analysis result over the Distance?